Source/AppAuthCore/OIDAuthorizationService.m

```diff
@@ -311,6 +311,19 @@ - (void)didFinishWithResponse:(nullable OIDEndSessionResponse *)response
 
 @implementation OIDAuthorizationService
 
+static OIDIDTokenValidator *_idTokenValidator;
+
++ (OIDIDTokenValidator *)idTokenValidator {
+  if (!_idTokenValidator) {
+    _idTokenValidator = [OIDIDTokenValidator new];
```

Comment:
This makes sense, but doesn't allow the integrator to explicitly set a nil validator in order to disable validation.
Would integrators be expected to supply their own validators which essentially perform a void operation in such a case?
Did you consider making this nullable?

Reply:
This is true. How could this be properly achieved in ObjC? With a `dispatch_once` block?

Reply:
Well, for an instance property I would suggest initializing the default in `init`. But as this is a class method, you'll need to go with overriding `+initialize` and initializing it there:
```
The runtime sends initialize() to each class in a program just before the class, or any class that inherits from it, is sent its first message from within the program. Superclasses receive this message before their subclasses.
```

Reply:
Thanks! I have applied the suggested changes, both the initialize and the nullability specifier, plus a comment.